refactor(Tree): Tree and TreeItem Indentation

ElliotTheHuman opened the pull request:
- Fixed border radius behavior and focus visible styling
- Added Asides to NavList and FieldPicker stories to mimic actual use cases
- Removed right padding from tree items and trees
- Deprecated branchFontWeight
- Deprecated assumeIconAlignment and forceLabelPadding props and removed stories
- Created NavTree and NavTreeItem components
- Resolved bug where tree item props landed on accordion DOM elements
- Deprecated TreeStyle

## Developer Checklist [ℹ️](https://github.com/looker-open-source/components/blob/main//CONTRIBUTING.md#developer-checklist)

- [ ] ♿️ Accessibility addressed
- [ ] 🌏 Localization & Internationalization addressed
- [ ] 🖼 Image Snapshot coverage
- [ ] 📚 Documentation updated
- [ ] 🧳 Bundle size impact addressed
- [ ] 🏁 Performance impacts addressed
- [ ] 👾 Browsers tested
  - [ ] Chrome
  - [ ] Firefox
  - [ ] Safari
  - [ ] IE11

lukelooker reviewed (commented):
Ok - quick review of snapshots shows a few issues:

- NavLIst/Mixed Navigation
  - `NavTree` - Text in "selected" `Tree` is no longer using key color (see `NavTree`)
  - `NavTreeItem` - text and icon no longer using key color  (see `My Awesome Item`)

- Border Radius appears to have lost radius and background (maybe just background is making radius invisible since background/foreground match?)

- `Tree: Disabled & Selected` - indicator is now black/tex5 instead of the previous lighter color (not exactly sure of value)

![image](https://user-images.githubusercontent.com/34253496/125573178-b79bdd5b-9551-43bf-a7dd-8ad7edf71544.png)

- `NavList` and tree-truncate examples are both making me wonder if we want to apply a small amount of padding to the right side of the truncate to prevent it from getting too close to the edge with ellipsis:

![image](https://user-images.githubusercontent.com/34253496/125572920-d44f8c7c-3a87-472b-9bb5-65302c25e284.png)

ElliotTheHuman commented on packages/components/src/NavList/NavList.story.tsx:
```diff
@@ -78,46 +78,36 @@ export const MixedNavigation = () => (
-        <Tree label="Folders" defaultOpen assumeIconAlignment>
-          <TreeItem truncate>
+        <NavTree label="Folders" defaultOpen>
+          <NavTreeItem description="description" detail="detail" selected>
+            My Awesome Item
+          </NavTreeItem>
```
I don't believe any child tree items in DiscoNav's example have icons so I removed the icon on the child tree item in our example. 

If there are, they we may need to introduce a new prop to prevent some child tree items from misaligning themselves relative to our normal alignment pattern.

ElliotTheHuman commented:
Alright, I _think_ I've fixed all the issues that popped up via snapshot discrepancies.

`NavList / Mixed Navigation`
- Fixed "selected" text color in a selected `NavTree` not being applied
- Fixed text and icon no longer using key color in `NavTreeItem`

`Tree: Disabled & Selected`
- Fixed indicator not being that lighter greyish color when disabled

`Border Radius`
- Border radius is working afaict, but the `dividers` prop wasn't working which I think resulted in the snapshot discrepancy. I went ahead and fixed that

lukelooker reviewed (commented):
Haven't made it all the way through this but wanted to give you feedback sooner than later since my schedule today is goofy. 

I pushed a PR trying to get as many of the bold text examples set back as before so I could see any intentional visual variances. 

Annoyingly some of the screenshots don't seem to be updating (re-running now to see if I can figure out where things went funny).

That said this is looking really good - most of my feedback is nits and/or asks for small additional explanations of some of the thornier pieces here.

lukelooker reviewed (approved):
Awesome refactor and A LOT easier to read through now - thanks for taking the time to do this as well as make sure snapshots were table to help make it feel stable :)
